tests/aclorch_ut.cpp

```diff
@@ -0,0 +1,1024 @@
+            { { TABLE_DESCRIPTION, "filter source IP" },
+                { TABLE_TYPE, TABLE_TYPE_L3 },
+                //            ^^^^^^^^^^^^^ L3 ACL
+                { TABLE_STAGE, TABLE_INGRESS },
+                // FIXME:      ^^^^^^^^^^^^^ only support / test for ingress ?
+                { TABLE_PORTS, "1,2" } } } });
+    // FIXME:                  ^^^^^^^^^^^^^ fixed port
+
+    orch->doAclTableTask(kvfAclTable);
+
+    // validate acl table ...
+
+    auto acl_table_oid = orch->getTableById(acl_table_id);
+    ASSERT_NE(acl_table_oid, SAI_NULL_OBJECT_ID);
+
+    const auto& acl_tables = orch->getAclTables();
+    auto it_table = acl_tables.find(acl_table_oid);
+    ASSERT_NE(it_table, acl_tables.end());
+
+    const auto& acl_table = it_table->second;
+
+    ASSERT_TRUE(validateAclTableByConfOp(acl_table, kfvFieldsValues(kvfAclTable.front())));
+    ASSERT_TRUE(validateLowerLayerDb(orch.get()));
+
+    // add rule ...
+    for (const auto& acl_rule_pkg_action : { PACKET_ACTION_FORWARD, PACKET_ACTION_DROP }) {
+
+        auto kvfAclRule = deque<KeyOpFieldsValuesTuple>({ { acl_table_id + "|" + acl_rule_id,
+            SET_COMMAND,
+            { { ACTION_PACKET_ACTION, acl_rule_pkg_action },
+
+                // if (attr_name == ACTION_PACKET_ACTION || attr_name == ACTION_MIRROR_ACTION ||
+                // attr_name == ACTION_DTEL_FLOW_OP || attr_name == ACTION_DTEL_INT_SESSION ||
+                // attr_name == ACTION_DTEL_DROP_REPORT_ENABLE ||
+                // attr_name == ACTION_DTEL_TAIL_DROP_REPORT_ENABLE ||
+                // attr_name == ACTION_DTEL_FLOW_SAMPLE_PERCENT ||
+                // attr_name == ACTION_DTEL_REPORT_ALL_PACKETS)
+                //
+                // TODO: required field (add new test cases for that ....)
+                //
+
+                { MATCH_SRC_IP, "1.2.3.4" },
+                { MATCH_DST_IP, "4.3.2.1" } } } });
+
+        // TODO: RULE_PRIORITY (important field)
+        // TODO: MATCH_DSCP / MATCH_SRC_IPV6 || attr_name == MATCH_DST_IPV6
+
+        orch->doAclRuleTask(kvfAclRule);
+
+        // validate acl rule ...
+
+        auto it_rule = acl_table.rules.find(acl_rule_id);
+        ASSERT_NE(it_rule, acl_table.rules.end());
+
+        ASSERT_TRUE(validateAclRuleByConfOp(*it_rule->second, kfvFieldsValues(kvfAclRule.front())));
+        ASSERT_TRUE(validateLowerLayerDb(orch.get()));
+
+        // delete acl rule ...
+
+        kvfAclRule = deque<KeyOpFieldsValuesTuple>({ { acl_table_id + "|" + acl_rule_id,
+            DEL_COMMAND,
+            {} } });
+
+        orch->doAclRuleTask(kvfAclRule);
+
+        // validate acl rule ...
+
+        it_rule = acl_table.rules.find(acl_rule_id);
+        ASSERT_EQ(it_rule, acl_table.rules.end());
+        ASSERT_TRUE(validateLowerLayerDb(orch.get()));
+    }
+}
+
+// When received ACL rule SET_COMMAND, orchagent can create corresponding ACL rule.
+// When received ACL rule DEL_COMMAND, orchagent can delete corresponding ACL rule.
```

Comment:
duplicated.

Reply:
I think no. One is SET, another is DEL command.

Reply:
Hi @lguohan, Is OK ?